logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/log/LogEntryDescriptor.java

```diff
@@ -150,16 +151,32 @@ public static void setKey(final MutableDirectBuffer buffer, final int offset, fi
     buffer.putLong(keyOffset(offset), key, Protocol.ENDIANNESS);
   }
 
-  public static int boolFlagsOffset(final int offset) {
-    return BOOL_FLAGS_OFFSET + offset;
+  public static int flagsOffset(final int offset) {
+    return FLAGS_OFFSET + offset;
   }
 
-  public static boolean isProcessed(final DirectBuffer buffer, final int offset) {
-    return (buffer.getByte(boolFlagsOffset(offset)) & 0x01) > 0;
+  public static byte getFlags(final DirectBuffer buffer, final int offset) {
+    return buffer.getByte(flagsOffset(offset));
   }
 
-  public static void markAsProcessed(final MutableDirectBuffer buffer, final int offset) {
-    buffer.putByte(boolFlagsOffset(offset), (byte) 1);
+  public static void setFlags(
+      final MutableDirectBuffer buffer, final int offset, final byte flags) {
+    buffer.putByte(flagsOffset(offset), flags);
+  }
+
+  public static boolean skipProcessing(final DirectBuffer buffer, final int offset) {
+    return (getFlags(buffer, offset) & SKIP_PROCESSING_BIT) != 0;
+  }
+
+  public static void setSkipProcessing(
+      final MutableDirectBuffer buffer, final int offset, final boolean skip) {
+    var flags = getFlags(buffer, offset);
```

Comment:
:thinking: Wondering why you did that now, as we discussed we haven't seen a need for now. This is not hard to add later when it is really needed. Please lets not add more code, which might be never used. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)